crates/rpc/rpc/src/debug.rs

```diff
@@ -733,12 +733,20 @@ where
 
                 let mut witness_record = ExecutionWitnessRecord::default();
 
-                let _ = block_executor
+                let output = block_executor
                     .execute_with_state_closure(&block, |statedb: &State<_>| {
                         witness_record.record_executed_state(statedb);
                     })
                     .map_err(|err| EthApiError::Internal(err.into()))?;
 
+                reth_ethereum_consensus::validate_block_post_execution(
```

Comment:
@klkvr we might want to use the actual `Consensus` implementation of the node here